From 6cc23aba0dbedcafe8162ff6ed906034b289d5a9 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Sat, 22 May 2021 22:42:36 +0300
Subject: [PATCH] [SW-43025] fwupd: secure i2c update fix

The original functionality patch had a copy/paste bug which disabled
security of the FW_UPD PCIe access.
Fix this by disabling the security toggling of the FW_UPD I2C access.

Change-Id: I7458e181a33366918376bb3d6f0ea1ad31707d72
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/preboot/module_fwupd.c b/preboot/module_fwupd.c
index dfdd53c..bf018be 100644
--- a/preboot/module_fwupd.c
+++ b/preboot/module_fwupd.c
@@ -254,7 +254,7 @@
 {
 	i2c_state.cmd_readl = i2c_fwupd_cmd_readl;
 	i2c_state.sts_writel = i2c_fwupd_sts_writel;
-	state.use_security = false;
+	i2c_state.use_security = false;
 	module_fwupd_send_status(&i2c_state, HL_UPD_STS_RST_OK, NULL);
 }
 
